src/components/SupportRequestModal.js

```diff
@@ -14,91 +14,98 @@ import validate from 'validate.js'
+          textInput({
+            placeholder: 'What should we call you?',
+            autoFocus: true,
+            value: nameEntered,
+            onChange: e => this.setState({ nameEntered: e.target.value })
+          })
+        ]),
+        Forms.requiredFormLabel('Type'),
+        h(Select, {
+          isMulti: false,
+          value: type,
+          onChange: ({ value }) => this.setState({ type: value }),
+          options: [{ value: 'question', label: 'Question' }, { value: 'bug', label: 'Bug' }, { value: 'feature_request', label: 'Feature Request' }]
+        }),
+        Forms.requiredFormLabel(`How can we help you${greetUser}?`),
+        textInput({
+          style: { borderBottomLeftRadius: 0, borderBottomRightRadius: 0, borderBottomStyle: 'dashed' },
+          placeholder: 'Enter a subject',
+          autoFocus: !!isSignedIn,
+          value: subject,
+          onChange: e => this.setState({ subject: e.target.value })
+        }),
+        h(TextArea, {
+          style: { height: 200, borderTopLeftRadius: 0, borderTopRightRadius: 0, borderTopStyle: 'dashed' },
+          placeholder: 'Enter a description',
+          value: description,
+          onChange: e => this.setState({ description: e.target.value })
+        }),
+        Forms.requiredFormLabel('Contact email'),
+        textInput({
           value: email,
+          placeholder: 'Enter your email address',
           onChange: e => this.setState({ email: e.target.value })
-        },
-        error: Utils.summarizeErrors(errors && errors.email)
-      }),
-      submitError && div({ style: { marginTop: '0.5rem', textAlign: 'right', color: colors.red[0] } }, [submitError]),
-      submitting && spinnerOverlay
-    ])
-  }
+        }),
+        submitError && div({ style: { marginTop: '0.5rem', textAlign: 'right', color: colors.red[0] } }, [submitError]),
+        submitting && spinnerOverlay
+      ])
+    }
 
-  async submit() {
-    const { onSuccess, ajax: { User }, authState: { profile: { firstName, lastName } } } = this.props
-    const { email, type, description, subject } = this.state
-    const name = `${firstName} ${lastName}`
-    const currUrl = window.location.href
+    async submit() {
+      const { onSuccess, authState: { isSignedIn, profile: { firstName, lastName } } } = this.props
+      const { email, type, description, subject, nameEntered } = this.state
+      const name = isSignedIn ? `${firstName} ${lastName}` : nameEntered
+      const currUrl = window.location.href
 
-    try {
-      this.setState({ submitting: true })
-      await User.createSupportRequest({ name, email, currUrl, type, description, subject })
-      onSuccess()
-    } catch (error) {
-      this.setState({ submitting: false })
-      reportError('Error submitting support request', error)
+      try {
+        this.setState({ submitting: true })
+        await Ajax().User.createSupportRequest({ name, email, currUrl, type, description, subject })
```

Comment:
I feel like I missed something, so sorry if this was already covered, but what was the reason for this?

Reply:
Not to worry, this was the result of an in-person conversation with Brett so you didn't miss anything in the PR that said why I changed this. There was a race condition that we hit when using ajaxCaller where sometimes we closed the modal before the Zendesk response was fully received. As a result, I got a console message that `User aborted request`. Changing to calling Ajax directly will avoid the race condition and leave the modal open slightly longer. If you have other ideas about how to avoid the race condition, or think it's worth looking into further, let me know.